gdx/src/com/badlogic/gdx/utils/IntFloatMap.java

```diff
@@ -320,9 +320,12 @@ public float remove (int key, float defaultValue) {
 		final int[] keyTable = this.keyTable;
 		final float[] valueTable = this.valueTable;
 		final float oldValue = valueTable[loc];
-		while ((key = keyTable[loc + 1 & mask]) != 0 && (loc + 1 & mask) != place(key)) {
+		int nl = (loc + 1 & mask);
+		while ((key = keyTable[nl]) != 0 && nl != place(key)) {
 			keyTable[loc] = key;
-			valueTable[loc] = valueTable[++loc & mask];
+			valueTable[loc] = valueTable[nl];
+			loc = nl;
+			nl = loc + 1 & mask;
```

Comment:
You don't need to increment loc anymore?

Reply:
This is weird code yeah, but nl starts with the value of (loc + 1 & mask), so we can assign that to loc to have it increment *and* wrap with mask. Then we need to increase nl again, also wrapping, so it still refers to the next slot.

Reply:
The bug before had to do with `++loc` incrementing but not wrapping, so when `valueTable[loc]` was accessed, it would be out of bounds. I'm wondering if some condition I didn't see was blocking that out of bounds case from occurring, because it never happened once over many tests and benchmarks. Still, it's better to have certainty that the bug can't happen or silently degrade behavior.

Reply:
@tommyettinger 

All, right, sorry for my review, next time I'll try to be more precise, especially reading my code better.